src/extractors/transComponent.ts

```diff
@@ -21,17 +21,18 @@ import { Config } from '../config';
 
 /**
  * Check whether a given JSXElement is a Trans component.
+ * @param transComponentNames: the list of possible Trans components with their source modules and
+ *        import names.
  * @param path: node path to check
  * @returns true if the given element is indeed a `Trans` component.
  */
 function isTransComponent(
+  transComponentNames: Config['transComponentNames'],
   path: BabelCore.NodePath<BabelTypes.JSXElement>,
 ): boolean {
   const openingElement = path.get('openingElement');
-  return referencesImport(
-    openingElement.get('name'),
-    'react-i18next',
-    'Trans',
+  return transComponentNames.some(({ sourceModule, importName }): boolean =>
```

Comment:
I feel concerned about how the complexity of `Trans` component extraction will evolve if we let the module resolution code here. I mean `TransComponent` extractor is already full of procedural code, recursions, corner-cases and subtilities. An alternative approach I was considering was to have a separate `customTransCompoent` extractor that relies on the `transComponent` extractor but skips the check. That relates to how other `t()` extractors work (like `translationRenderProp` for instance), and it may slightly reduce complexity by improving separation of concerns.

Reply:
I believe that there are two main features now:
- checking whether a node is to be extracted
- actually extracting

The first one is contained in a single function and we could easily group the second in a single function to be exported if it were to be used in another extractor.

However I don't see which other extractor you have in mind as they would basically be the same, just acting with a different config. I think it's different (for now) in `t` case, as the context in which you extract from t might give you different indications.

Anyway happy to refactor in 2 files if you think it's important. If so please confirm the way you would like to activate one extractor over the other (maybe it's related to your wish of having a default value being an empty array).